File: languages/java/src/main/java/com/bloock/sdk/entity/authenticity/BjjSigner.java
package com.bloock.sdk.entity.authenticity;

import com.bloock.sdk.bridge.proto.AuthenticityEntities;
import com.bloock.sdk.bridge.proto.AuthenticityEntities.SignerAlg;

public class BjjSigner implements Signer {
    AuthenticityEntities.SignerAlg alg;
    SignerArgs args;

    public BjjSigner(SignerArgs args) {
        this.alg = SignerAlg.BJJ;
        this.args = args;
    }

    @Override
    public AuthenticityEntities.Signer toProto() {
        AuthenticityEntities.Signer.Builder builder = AuthenticityEntities.Signer.newBuilder().setAlg(this.alg);

        if (this.args.localKey != null) {
            builder.setLocalKey(this.args.localKey.toProto());
        }

        if (this.args.managedKey != null) {
            builder.setManagedKey(this.args.managedKey.toProto());
        }

        if (this.args.commonName != null) {
            builder.setCommonName(this.args.commonName);
        }

        return builder.build();
    }
}

File: languages/php/Bloock/Entity/Authenticity/BjjSigner.php
<?php

namespace Bloock\Entity\Authenticity;

use Bloock\SignerAlg;

class BjjSigner implements Signer
{
    public int $alg;
    public SignerArgs $args;

    public function __construct(SignerArgs $args)
    {
        $this->alg = SignerAlg::BJJ;
        $this->args = $args;
    }

    public function toProto(): \Bloock\Signer
    {
        $s = new \Bloock\Signer();
        $s->setAlg($this->alg);

        if ($this->args->localKey != null) {
            $s->setLocalKey($this->args->localKey->toProto());
        }

        if ($this->args->managedKey != null) {
            $s->setManagedKey($this->args->managedKey->toProto());
        }

        if ($this->args->managedCertificate != null) {
            $s->setManagedCertificate($this->args->managedCertificate->toProto());
        }

        if ($this->args->commonName != null) {
            $s->setCommonName($this->args->commonName);
        }

        return $s;
    }
}
